dashboard/test/controllers/plc/enrollment_evaluations_controller_test.rb

```diff
@@ -0,0 +1,103 @@
+    @task6 = create(:plc_task, name: 'Not getting your favorite color right', plc_learning_module: @module1)
+
+    @question1 = create(:plc_evaluation_question, question: 'What is your name', plc_course: @course)
+    @question2 = create(:plc_evaluation_question, question: 'What is your quest', plc_course: @course)
+    @question3 = create(:plc_evaluation_question, question: 'What is your favorite color?', plc_course: @course)
+    @question4 = create(:plc_evaluation_question, question: 'What is the capital of Assyria?', plc_course: @course)
+    @question5 = create(:plc_evaluation_question, question: 'What is the airspeed velocity of an unladen swallow', plc_course: @course)
+
+    @answer1_1 = create(:plc_evaluation_answer, answer: 'Sir Lancelot', plc_evaluation_question: @question1, plc_task: @task1)
+    @answer1_2 = create(:plc_evaluation_answer, answer: 'Sir Robin', plc_evaluation_question: @question1, plc_task: @task1)
+    @answer1_3 = create(:plc_evaluation_answer, answer: 'Sir Galahad', plc_evaluation_question: @question1, plc_task: @task1)
+    @answer1_4 = create(:plc_evaluation_answer, answer: 'King Arthur', plc_evaluation_question: @question1, plc_task: @task1)
+    @answer1_5 = create(:plc_evaluation_answer, answer: 'Sir Edgecase', plc_evaluation_question: @question1, plc_task: nil)
+
+    @answer2_1 = create(:plc_evaluation_answer, answer: 'I seek the Grail', plc_evaluation_question: @question2, plc_task: @task2)
+    #Not all answers are guaranteed to have associated tasks
+    @answer2_2 = create(:plc_evaluation_answer, answer: 'I seek something else Grail', plc_evaluation_question: @question2, plc_task: nil)
+
+    @answer3_1 = create(:plc_evaluation_answer, answer: 'Blue', plc_evaluation_question: @question3, plc_task: @task3)
+    @answer3_2 = create(:plc_evaluation_answer, answer: 'Yellow - no, blue', plc_evaluation_question: @question3, plc_task: @task6)
+    @answer3_3= create(:plc_evaluation_answer, answer: 'No preference', plc_evaluation_question: @question3, plc_task: nil)
+
+    @answer4_1 = create(:plc_evaluation_answer, answer: 'I don\t know that', plc_evaluation_question: @question4, plc_task: @task4)
+    @answer4_2 = create(:plc_evaluation_answer, answer: 'Nineveh', plc_evaluation_question: @question4, plc_task: nil)
+
+    @answer5_1 = create(:plc_evaluation_answer, answer: 'What do you mean, an African or European Swallow', plc_evaluation_question: @question5, plc_task: @task5)
+    @answer5_2 = create(:plc_evaluation_answer, answer: '15 m/s', plc_evaluation_question: @question5, plc_task: nil)
+
+    @user = create :admin
+    sign_in(@user)
+
+    @plc_enrollment = create(:plc_user_course_enrollment, user: @user, plc_course: @course)
+  end
+
+  test "perform_evaluation retrieves all questions and answers" do
+    get :perform_evaluation, enrollment_id: @plc_enrollment.id
+    questions = @controller.instance_variable_get(:@questions)
```

Comment:
you can get instance variables in controller tests with eg. `questions = assigns(:questions)`